File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
    
        arr = list(range(0,n))
        print (arr)
        
        def helper(n):
            if(n==arr[n]):
                return n
            return helper(arr[n])
            
        for i, j in edges:
            arr[j]=helper(i)
        # print (arr)
            
        # print (arr)
        res=[]
        for idx, x in enumerate(arr):
            if(x==idx):
                res.append(x)
                
        return res

File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.js
/**
 * @param {number} n
 * @param {number[][]} edges
 * @return {number[]}
 */
var findSmallestSetOfVertices = function(n, edges) {
    let arr=Array.from({length: n}, (_,i)=>i)
    // edges.sort((a,b)=>b[1]-a[1])
    function helper(n){
        if(n===arr[n])
            return n
        return helper(arr[n])
        
    }
    for(let [i, j] of edges){
        arr[j]=helper(i)
    }
    // console.log(arr)
    let res=[]
    for(let i=0;i<n;i++){
        if(arr[i]===i)
            res.push(i)
    }
    return res
};